subworkflows/local/taxid_bam_fasta.nf

```diff
@@ -0,0 +1,109 @@
+//
+// Prepare an individual BAM/FASTA file for each pathogen with mapped reads
+//
+
+include { SUBSET_BAM as SUBSET_BAM_PASS                     } from '../../modules/local/subset_bam'
+include { SUBSET_BAM as SUBSET_BAM_FAIL                     } from '../../modules/local/subset_bam'
+include { SAMTOOLS_SORT as SAMTOOLS_SORT_PASS               } from '../../modules/nf-core/samtools/sort/main'
+include { SAMTOOLS_SORT as SAMTOOLS_SORT_FAIL               } from '../../modules/nf-core/samtools/sort/main'
+include { SAMTOOLS_INDEX                                    } from '../../modules/nf-core/samtools/index/main'
+include { SAMTOOLS_IDXSTATS                                 } from '../../modules/nf-core/samtools/idxstats/main'
+include { SAMTOOLS_FASTA                                    } from '../../modules/nf-core/samtools/fasta/main'
+
+workflow TAXID_BAM_FASTA {
+    take:
+    bam               // Channel: [ val(meta), path(bam) ]
+    bai               // Channel: [ val(meta), path(bai) ]
+    accession2taxid   // Channel: path(accession2taxid)
+    min_read_counts   // Value: minimum number of reads to keep a BAM file
+
+    main:
+    ch_versions       = Channel.empty()
+    ch_multiqc_files  = Channel.empty()
+
+    // Combine BAM and BAI files
+    input_bam = bam.combine(bai, by: 0)
```

Comment:
I am wondering if `join` should be used as an operator instead. Is there a chance here that the files can be mixed if they are not combined based on an identifier?

Reply:
Updated.